pykusto/_src/expressions.py

```diff
@@ -53,10 +53,12 @@ def as_subexpression(self) -> KQL:
     def get_type(self) -> '_StringExpression':
         return _StringExpression(KQL(f'gettype({self.kql})'))
 
-    def __hash__(self) -> '_StringExpression':
+    def __hash__(self, mod: int = None) -> '_StringExpression':
```

Comment:
Allow using a numeric expression, not only a literal number (but in case of a literal make sure it's an integer)